cmd/storer2elastic/dataprocessor/dataProcessor.go

```diff
@@ -115,10 +124,15 @@ func (dp *dataProcessor) processData(persistedData storer2ElasticData.RoundPersi
 	metaBlock, _ := metaPersistedData.Header.(*block.MetaBlock)
 	dp.tpsBenchmarkUpdater.IndexTPSForMetaBlock(metaBlock)
 
-	for _, shardData := range persistedData.ShardHeaders {
-		err = dp.indexData(shardData)
-		if err != nil {
-			log.Warn("error indexing shard header", "error", err)
+	for _, shardDataForShard := range persistedData.ShardHeaders {
+		for _, shardData := range shardDataForShard {
+			err = dp.indexData(shardData)
+			if err != nil {
+				log.Warn("error indexing shard header",
```

Comment:
this function returns a `bool`. added `return false` when an error occurs so the tools won't continue fetching data from the `dataReplayer` and it will exit. also added in other places where the error was only logged